cluster/manager.go

```diff
@@ -266,6 +268,30 @@ func (c *ClusterManager) GetData() (map[string]*api.Node, error) {
 	return nodes, nil
 }
 
+func (c *ClusterManager) nodeIdFromIp(idIp string) (string, error) {
+	// Caller's responsibility to lock the access to the NodeCache.
+	for _, n := range c.nodeCache {
+		if n.DataIp == idIp || n.MgmtIp == idIp {
+			logrus.Infof("Node IP: " + idIp + " maps to ID: " + n.Id)
```

Comment:
nit: may be you want to remove his log line as it will log for every http request which makes call to this API.